rest_framework/serializers.py

```diff
@@ -663,7 +668,10 @@ def to_internal_value(self, data):
         ret = []
         errors = []
 
-        for item in data:
+        for idx, item in enumerate(data):
+            if hasattr(self, 'instance') and self.instance and \
+                    len(self.instance) > idx:
```

Comment:
I do not know if this repo is using backlashes or parenthesis to wrap long lines:
```suggestion
            if (
                hasattr(self, 'instance')
                and self.instance
                and len(self.instance) > idx
            ):
```